src/linked_list.cr

```diff
@@ -199,6 +271,34 @@ class LinkedList(A)
     end
   end
 
+  def [](index : Int32)
+    return nil if @head.nil?
```

Comment:
Why not rely on `each_node` instead?

Reply:
How do you mean?

Reply:
Could you iterate over the list using `each_node` and maintain a separate counter? When the counter equals the requested index, then return early from the block with the value of the node.

Reply:
oh I get it, let me try

Reply:
should be good now, unit tests still pass

Reply:
Great, I can land this for you later tonight once I'm done with work.